JASP-Engine/JASP/R/exploratoryfactoranalysis.R

```diff
@@ -82,18 +82,24 @@ ExploratoryFactorAnalysis <- function(jaspResults, dataset, options, ...) {
         return(gettext("Data not valid: variance is zero in each row"))
       }
     },
-    # check whether all variables correlate with each other
+    # Check for correlation anomalies
     function() {
-      allCorr <- 0
-      nVar <- ncol(dataset)
-      for (i in 1:(nVar-1)) {
-        for (j in (i+1):nVar) {
-          thisCor <- cor(dataset[,i],dataset[,j])
-          if(!is.na(thisCor) && thisCor == 1) allCorr <- allCorr + 1
-        }
+      P <- ncol(dataset)
+      
+      # check whether a variable has too many missing values to compute a 
+      # correlation
+      if (any(is.na(S))) {
```

Comment:
Unless I'm missing it `S` is used before it's defined

Reply:
wow that's stupid of me. I'll fix ASAP